boa/src/syntax/lexer/cursor.rs

```diff
@@ -283,11 +292,13 @@ where
     /// Retrieves the next ASCII checked character.
     #[inline]
     fn next_ascii(&mut self) -> io::Result<Option<u8>> {
-        let next_byte = self.iter.next().transpose()?;
-
-        match next_byte {
-            Some(next) if next <= 0x7F => Ok(Some(next)),
-            None => Ok(None),
+        match self.next_char() {
+            Ok(Some(chr)) if chr.is_ascii() => {
+                let mut buf = [0u8; 4];
+                chr.encode_utf8(&mut buf);
+                Ok(Some(buf[0]))
```

Comment:
```suggestion
                Ok(Some(chr as u8))
```
Couldn't we just cast to u8 since we check if its ascii, this should be faster than calling `encode_utf8`

Reply:
Sounds great! I've made this change.

Reply:
Besides, I can look into the invalid code point issue. Is there already an issue/PR for it?

Reply:
> Besides, I can look into the invalid code point issue. Is there already an issue/PR for it?

There is an issue https://github.com/boa-dev/boa/issues/778, but not a PR to fix